### PR TITLE
fix(semantic): report errors for missing class method impls

### PR DESCRIPTION
**Pull request by camc314**

### CODE REVIEW

**Comment by camc314** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8082?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#8082** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8082?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8082?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#8081** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8081?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=8082).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: tasks/coverage/snapshots/parser_typescript.snap [hunk 1]
```diff
@@ -4792,6 +4766,24 @@ Expect to Parse: tasks/coverage/typescript/tests/cases/conformance/moduleResolut
...
+Expect to Parse: tasks/coverage/typescript/tests/cases/conformance/nonjsExtensions/declarationFileForHtmlImport.ts
+
+  × Function implementation is missing or not immediately following the declaration.
+    ╭─[typescript/tests/cases/conformance/nonjsExtensions/declarationFileForHtmlImport.ts:11:5]
+ 10 │ export class HTML5Element extends HTMLElement {
+ 11 │     connectedCallback(): void;
+    ·     ─────────────────
+ 12 │ }
+    ╰────
 Expect to Parse: tasks/coverage/typescript/tests/cases/conformance/salsa/plainJSRedeclare3.ts
```
Comment (camc314) [inline tasks/coverage/snapshots/parser_typescript.snap:4787]: this is correct.

`@filename` is speficied to be `component.d.html`, but we are checking it as if it's TS

https://github.com/microsoft/typescript/blob/89e928e8b41368bb0888c6d399854c53a2dbf3d4/tests/cases/conformance/nonjsExtensions/declarationFileForHtmlFileWithinDeclarationFile.ts#L3-L16

Comment (Boshen) [reply tasks/coverage/snapshots/parser_typescript.snap:4787]: Where's the documentation for considering `.d.html.ts` a ts definition file lol, we can follow up by setting such files as `.d.ts`.

Comment (camc314) [reply tasks/coverage/snapshots/parser_typescript.snap:4787]: Ok, this is stupid, but here is the commit

https://github.com/microsoft/TypeScript/commit/89e928e8b41368bb0888c6d399854c53a2dbf3d4#diff-5f0fce51104cdb1193245a158673eaf24b263dd4ac2bd5f5a26aa57dfa59e0a8R216

on mobile so not sure if it's linked correctly, but look at

src/harness/compilerImpl.ts#L216

Comment (camc314) [reply tasks/coverage/snapshots/parser_typescript.snap:4787]: in the test runner, probably need to split the files at any @ filename and run each of them through the parser to make the tests more accurate

**Comment by Boshen** [comment]
### Merge activity

* **Dec 23, 8:35 PM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

**Comment by Boshen** [comment]
I'll add some follow up PRs.

**Pull request merged**